api/src/main/java/com/eleks/academy/pharmagator/scheduler/Scheduler.java

```diff
@@ -14,19 +20,50 @@
 
 @Slf4j
 @Component
-@Profile("!test")
 @RequiredArgsConstructor
 public class Scheduler {
 
     private final List<DataProvider> dataProviderList;
+    private final PriceRepository priceRepository;
+    private final PharmacyRepository pharmacyRepository;
+    private final MedicineRepository medicineRepository;
+    private final ModelMapper modelMapper;
 
-    @Scheduled(fixedDelay = 1, timeUnit = TimeUnit.MINUTES)
+    @Scheduled(fixedDelay = 100, timeUnit = TimeUnit.MINUTES)
     public void schedule() {
         log.info("Scheduler started at {}", Instant.now());
-        dataProviderList.stream().flatMap(DataProvider::loadData).forEach(this::storeToDatabase);
+        dataProviderList.stream()
+                .flatMap(DataProvider::loadData)
+                .forEach(this::storeToDatabase);
+        log.info("Scheduler finished at {}", Instant.now());
     }
 
     private void storeToDatabase(MedicineDto dto) {
-        log.info(dto.getTitle() + " - " + dto.getPrice());
+        Medicine medicine = modelMapper.map(dto, Medicine.class);
+        Price price = modelMapper.map(dto, Price.class);
+
+        Pharmacy pharmacy = new Pharmacy();
+        pharmacy.setName(dto.getPharmacy());
+        Pharmacy pharmacyFromDb = pharmacyRepository.findFirstByName(pharmacy.getName());
+        if (pharmacyFromDb != null) {
```

Comment:
Why do You need to re-save pharmacy?
In most cases it will remain the same, so? all You need is to skip pharmacy processing if it is already saved.